src/compose/application-manager.ts

```diff
@@ -999,21 +1004,55 @@ export async function getState() {
 		createdAt,
 		...svc
 	} of servicesToReport) {
-		state[appUuid] = {
-			...state[appUuid],
+		const app = state[appUuid] || {
```

Comment:
Should this be
```suggestion
		const app = state[appUuid] ?? {
```
?

(also if you update to the latest balena-lint it'll ask this question automatically, but I find it very important to distinguish between when `||` is the intent and `??` is the intent as they mean different things and knowing the true intent of the code really helps at least me to understand it - and it's far easier to know that intent at the time of writing rather than trying to work it out later because if you make a mistake then it'll be a real problem)